#! /bin/sh -e
## 61_uaccess_breakage.dpatch by Thiemo Seufer <ths@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Work around uaccess breakage.

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

--- build-flavour-sb1-swarm-bn/fs/compat_ioctl.c.old	2005-10-13 02:14:34.000000000 +0200
+++ build-flavour-sb1-swarm-bn/fs/compat_ioctl.c	2005-10-13 02:14:53.000000000 +0200
@@ -1925,7 +1925,7 @@ struct floppy_struct32 {
 	unsigned char	rate;
 	unsigned char	spec1;
 	unsigned char	fmt_gap;
-	const compat_caddr_t name;
+	compat_caddr_t	name;
 };
 
 struct floppy_drive_params32 {
